python/rmm/_version.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__version__ = "23.12.00"
```

Comment:
<!-- Polish comments are like a suggestion, where there is nothing necessarily wrong with the relevant content, there's just some ways to immediately improve the quality.  -->
**polish:** ‏Indicate that this file is "auto-generated":

```suggestion
# These fields are updated by release/package scripts.
# In particular:
# - ci/build_python.sh
# - ci/build_wheel.sh
# - ci/release/update-version.sh
__version__ = "23.12.00"
```

WDYT?

Reply:
I'm okay either way here. Indicating it's generated is fine, but I wouldn't list the specific scripts. That's fragile and those files aren't necessarily shipped with the package containing this.

Reply:
I think I would actually prefer not to indicate this. The reason is that while it is true that they are updated by release scripts, in practice most people who actually read this code will be reading the un-updated source code with these literal values. The update is more a function of the release process than it is of the source itself. OTOH I could see prepending this comment during the release process instead, which would be more accurate and appropriate. WDYT? Is that worthwhile?

Reply:
There's a lot of hidden-ish steps... It's gradually improving and becoming less opaque. In general you can assume that the build scripts in the `ci/` directory represent how releases are built. The actual release process is simply a matter of copying those files out somewhere else. The build scripts do have a bit of magic to them though, so simply running a `pip install .` or a `conda-build /path/to/recipe` won't just work out of the box. Over time we can hopefully ameliorate some of that complexity.